### ISSUE TITLE
Fix dynamic linking on Windows

### ISSUE THREAD

**Issue by IvanUkhov** [blas-lapack-rs/openblas-src#10]
The provider apparently doesn’t work on Windows. See [here](https://github.com/stainless-steel/blas/issues/17).

**Comment by scharris** [comment]
It actually also doesn't work on Linux either if there are any spaces in the path to the project directory.

To summarize known issues and solutions with build.rs:

- Installing Msys2 and the dev tools via its provided pacman tool ("pacman -S make perl gcc gfortran") fixes the linker problem, and this could have been caused by mixing the wrong mingw and msys installations on my system (as mentioned that ecosystem is a mess).
- Next on Windows there was an error in the "make install" step, because openblas's scripts cannot handle Windows style paths with backslashes (demonstrable without Cargo/Rust involved).  This is not surprising since openblas is using a unix-like environment (Msys and/or MingW) to build with unix tools on Windows.  The fix is to convert backslashes to forward slashes in the destination path arg provided to make in buiild.rs.
- Openblas doesn't properly quote arguments in its build scripts somewhere, so in build.rs, `DESTDIR={}` needs to be `DESTDIR='{}'`, which fixes the problem of the project path containing spaces, on both Windows and Linux. A similar change was done for the link search path.

The proposed fix (all in build.rs) is [here](https://github.com/scharris/openblas-provider/commit/e281c83356d9a4c566e6dc30a22dddd5bd73f9ae?diff=unified) (to see code changes unchecking comments helps)

**Comment by IvanUkhov** [comment]
Could you please open a pull request for your changes with respect to the latest version of the build script (I did some cleanup today)? I think the interception point for replacing slashes should be [this](https://github.com/cmr/openblas-provider/blob/master/build.rs#L23) line. I guess you can replace right there without auxiliary functions.

Regarding testing, I’d like to make sure that it works on OS X and Linux as before, so I’m setting up Travis CI for this repository (just waiting for a response from @cmr). Regarding Windows, AppVeyor [doesn’t seem](https://github.com/appveyor/ci/issues/733) to support MSYS2 _yet_. What it currently supports can be inferred from [here](https://github.com/cmr/openblas-provider/blob/master/appveyor.yml#L10), I guess.

**Comment by scharris** [comment]
If the replacement is done at the top like that then it won't be a pathbuf anymore there (it will be just a string) unlike the source var, and then the join on it for the rustc-link-search would have to be just a string join not a pathbuf join (a bit fragile since pathbuf handles doubled slashes etc).  I guess that's OK, I just want to make sure that's what you want.

If you want to make the changes yourself then I can try them on my windows machine - if not just let me know I should be able to get to this soon, probably later today.

**Comment by IvanUkhov** [comment]
Yes, you’re absolutely right. Perhaps converting back to `PathBuf` right after replacing?

I just thought it would be nice to have your contribution reflected in the commit history. I can surely apply the changes myself if it’s OK with you.

**Comment by scharris** [comment]
Hmm that idea seems to work here on Windows 7, but if you don't mind I'd like to test on Windows 10 which shows different behavior sometimes - will do that sometime tomorrow morning when I'm back at that machine.   

I noticed that on Windows 7 with MSys2 it isn't actually requiring the backslashes to be replaced when the single quotes are present around the path (that would be nice as the quotes are needed anyway for pathnames containing spaces) - but I need to test on Windows 10 to make sure (I can't remember if I tested that).

PS: Thanks for the thought of including me in the history - whatever's most convenient is fine with me (but let me test Win 10 first!).
